### PR TITLE
feat: separated type providers

### PR DESCRIPTION
**Pull request by Bram-dc**
I modified the type provider to work correctly with the changes in this PR, which has been added to fastify v5: 

[PR](https://github.com/fastify/fastify/pull/5427)
[Discussion](https://github.com/fastify/fastify/discussions/5311)

### CODE REVIEW

**Comment by thelinuxlich** [comment]
+1, this is important to migrate to Fastify v5

**Comment by Bram-dc** [comment]
@turkerdev fastify 5.0.0 has been released recently, could you take a look?

**Comment by kibertoad** [comment]
thank you! will review today

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "prepublishOnly": "npm run build"
   },
   "peerDependencies": {
-    "fastify": "^4.0.0",
+    "fastify": "^5.0.0-alpha.3",
```
Comment (kibertoad) [inline package.json:22]: can you bump this to a final release?

Comment (Bram-dc) [reply package.json]: Done!

**Comment by kibertoad** [comment]
@Bram-dc thank you! will merge after creating a v4 branch

**Comment by kibertoad** [comment]
@Bram-dc CI is failing

**Comment by Bram-dc** [comment]
Should be fixed now

**Comment by kibertoad** [comment]
@Bram-dc still failed

**Comment by Bram-dc** [comment]
> @Bram-dc still failed

Could this be an issue not related this PR? Do the CI jobs on the main branch work?

**Comment by kibertoad** [comment]
please drop it from the matrix then

**Comment by Bram-dc** [comment]
> Fastify v5 will only support Node.js v20+. If you are using an older version of Node.js, you will need to upgrade to a newer version to use Fastify v5. https://fastify.dev/docs/latest/Guides/Migration-Guide-V5/

**Comment by kibertoad** [comment]
fastify is node 20+ now

**Comment by kibertoad** [comment]
@Bram-dc could you please resolve conflicts?

**Comment by kibertoad** [comment]
@Bram-dc can you rebase to have proper diff?

**Comment by Bram-dc** [comment]
Can we merge anyway? It an issue with the node version again.

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
thanks a lot!

**Pull request merged**